packages/oid4vci-rest-client/src/agent/OID4VCIRestClient.ts

```diff
@@ -0,0 +1,65 @@
+import { fetch } from 'cross-fetch'
+import {
+  IOID4VCIRestClient,
+  IVCIClientCreateOfferUriRequest,
+  IVCIClientCreateOfferUriRequestArgs,
+  IVCIClientCreateOfferUriResponse,
+} from '../types/IOID4VCIRestClient'
+import Debug from 'debug'
+import { IAgentPlugin } from '@veramo/core'
+
+const debug = Debug('sphereon:ssi-sdk.oid4vci-rest-client')
+
+/**
+ * @beta
+ */
+export class OID4VCIRestClient implements IAgentPlugin {
+  readonly methods: IOID4VCIRestClient = {
+    vciClientCreateOfferUri: this.vciClientCreateOfferUri.bind(this),
+  }
+
+  private readonly baseUrl?: string
+
+  constructor(args?: { baseUrl?: string; definitionId?: string }) {
+    if (args?.baseUrl) {
+      this.baseUrl = args.baseUrl
+    }
+  }
+
+  private async vciClientCreateOfferUri(args: IVCIClientCreateOfferUriRequestArgs): Promise<IVCIClientCreateOfferUriResponse> {
+    if (!args.credentials) {
+      throw new Error("Can't generate the credential offer uri without credentials param present.")
+    }
+    const baseUrl = this.checkBaseUrlParameter(args.baseUri)
+    const request: IVCIClientCreateOfferUriRequest = {
+      credentials: args.credentials,
+    }
+    if (args.grants) {
+      request['grants'] = args.grants
+    }
+    const url = this.uriWithBase(`webapp/credential-offers`, baseUrl)
+    debug(`OID4VCIRestClient is going to send request: ${JSON.stringify(request)} to ${url}`)
+    const origResponse = await fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(request),
+    })
+    return await origResponse.json()
+  }
+
+  private uriWithBase(path: string, baseUrl?: string): string {
```

Comment:
suggestion: uriWithBaseFrom

Reply:
I am fine with this name, given it will result in a URL with the baseURL included, either from global configured baseUrl or from the Argument. Everything should however be called Url and not Uri, as we are only talking about the Url of the agent, which will never be anything other than a URL